pkg/api/rest/client.go

```diff
@@ -0,0 +1,69 @@
+package rest
+
+import (
+	"net/url"
+	"strings"
+
+	"github.com/pingcap/ticdc/pkg/httputil"
+)
+
+// RESTInterface includes a set of operations to interact with cdc REST apis
+type RESTInterface interface {
+	Verb(verb string) *Request
+	Post() *Request
+	Put() *Request
+	Get() *Request
+	Delete() *Request
+}
+
+type RESTClient struct {
+	// base is the root URL for all invocations of the client.
+	base *url.URL
+
+	// versionedAPIPath is a http url prefix with api version. eg. /api/v1.
+	versionedAPIPath string
+
+	// Client is a wrapped http client.
+	Client *httputil.Client
+}
+
+// NewRESTClient creates a new RESTClient.
+func NewRESTClient(baseURL *url.URL, versionedAPIPath string, client *httputil.Client) (*RESTClient, error) {
+	base := *baseURL
```

Comment:
What is the motivation to do the dereference `*`?

Reply:
If it is to avoid accidental modification by pointer, can you add it in comment?

Reply:
The dereference can be eliminated. I will do it later.